daemon/api_snapshots_test.go

```diff
@@ -68,6 +72,48 @@ func (s *snapshotSuite) TestSnapshotMany(c *check.C) {
 	c.Check(res.Affected, check.DeepEquals, inst.Snaps)
 }
 
+func (s *snapshotSuite) TestSnapshotManyOptionsFull(c *check.C) {
+	defer daemon.MockSnapshotSave(func(s *state.State, snaps, users []string,
+		options map[string]*snap.SnapshotOptions) (uint64, []string, *state.TaskSet, error) {
+		c.Check(snaps, check.HasLen, 2)
+		c.Check(options, check.HasLen, 2)
+		c.Check(options, check.DeepEquals, map[string]*snap.SnapshotOptions{
+			"foo": {Exclude: []string{"foo-path-1", "foo-path-2"}},
+			"bar": {Exclude: []string{"bar-path-1", "bar-path-2"}},
+		})
+		t := s.NewTask("fake-snapshot-2", "Snapshot two")
+		return 1, snaps, state.NewTaskSet(t), nil
+	})()
+
+	inst := daemon.MustUnmarshalSnapInstruction(c, `{"action": "snapshot", "snaps": ["foo", "bar"],
+	"snapshot-options": {"foo": {"exclude":["foo-path-1", "foo-path-2"]}, "bar":{"exclude":["bar-path-1", "bar-path-2"]}}}`)
+
+	st := s.d.Overlord().State()
+	st.Lock()
+	res, err := inst.DispatchForMany()(inst, st)
+	st.Unlock()
+	c.Assert(err, check.IsNil)
+	c.Check(res.Summary, check.Equals, `Snapshot snaps "foo", "bar"`)
+	c.Check(res.Affected, check.DeepEquals, inst.Snaps)
+}
+
+func (s *snapshotSuite) TestSnapshotManyError(c *check.C) {
```

Comment:
Why add this test? This doesn't seem related to the new code

Reply:
maybe it's covering a path that was not covered before?

Reply:
Indeed, to cover the error path that was not covered before.